### ISSUE TITLE
Create separate providers for each media type supported by the image provider

### ISSUE THREAD

**Issue by oparoz** [owncloud/core#15071]
It started with
```
public function getMimeType() {
	return '/image\/.*/';
}
```
and we're already at
```
public function getMimeType() {
	return '/image\/(?!x-dcraw$)(?!tiff$)(?!svg.*).*/';
}
```
It needs to be changed to something like this, with one class per media type.
```
public function getMimeType() {
	return '/image\/png/';
}
```
etc.

**Comment by nickvergessen** [comment]
Just make it `return '/image\/(png|jpeg|...)/';` for now, so it doesn't match tiff and svg anymore

**Comment by oparoz** [comment]
But this goes directly against the rule which says that each mime type needs to be in its own class and we can't create new classes.
So I can do it if @DeepDiver1975 says it's OK.

**Comment by nickvergessen** [comment]
The split it up into multiple classes.

**Comment by oparoz** [comment]
Its true that the image provider is not going anywhere, so it should be "legal" to split it.

Any objection @georgehrke ?

**Comment by georgehrke** [comment]
No objections from my side.

**Comment by oparoz** [comment]
Thanks

**Comment by DeepDiver1975** [comment]
@oparoz @nickvergessen is this fixed with #15346? THX

**Comment by nickvergessen** [comment]
yes
